tools/mtmd/clip.cpp

```diff
@@ -3192,7 +3204,30 @@ struct clip_model_loader {
         }
     }
 
-    void alloc_compute_meta(clip_ctx & ctx_clip) {
+    void warmup(clip_ctx & ctx_clip) {
+        if (ctx_clip.flash_attn_type == LLAMA_FLASH_ATTN_TYPE_AUTO) {
+            // try to enable flash attention to see if it's supported
+            ctx_clip.flash_attn_type = LLAMA_FLASH_ATTN_TYPE_ENABLED;
+            bool supported = alloc_compute_meta(ctx_clip);
+            if (!supported) {
+                LOG_WRN("%s: flash attention not supported, memory usage will increase\n", __func__);
+                // TODO: maybe log more details about why flash attention is not supported
```

Comment:
@ggerganov I implemented a simple solution to auto-enable flash attn only when the backend support it. Probably we should make this `LOG_WRN` to be more prominent. Also, which kind of info do you think should be displayed here?

Some users potentially already using models with shapes not supported by GPU flash attn. Falling back to CPU will suddenly make it very slow and thus not a good UX overall. The auto mode + prominent is a better solution as also it encourage users to "voluntary" report certain info back to us - less forcefully for them.

Reply:
>  Also, which kind of info do you think should be displayed here?

We can print the actual tensor (shape, strides, types) for which FA is not supported.